domain/ide/workspace/folder.go

```diff
@@ -468,18 +532,26 @@ func (f *Folder) IssuesForRange(filePath string, requestedRange snyk.Range) (mat
 }
 
 func (f *Folder) ClearDiagnostics() {
+	// send global cache evictions
 	f.documentDiagnosticCache.Range(func(key string, _ []snyk.Issue) bool {
-		// we must republish empty diagnostics for all files that were reported with diagnostics
-		f.notifier.Send(lsp.PublishDiagnosticsParams{
-			URI:         uri.PathToUri(key),
-			Diagnostics: []lsp.Diagnostic{},
-		})
-		f.documentDiagnosticCache.Delete(key)
+		f.ClearDiagnosticsFromGlobalCache(key)
 		return true
 	})
+	f.documentDiagnosticCache.Clear()
+
+	// let scanner-local cache handle its own stuff
+	if cacheProvider, isCacheProvider := f.scanner.(snyk.CacheProvider); isCacheProvider {
+		issues := cacheProvider.Issues()
+		for path := range issues {
+			if f.Contains(path) {
+				cacheProvider.ClearIssues(path)
+			}
+		}
+	}
 }
 
 func (f *Folder) ClearDiagnosticsByIssueType(removedType product.FilterableIssueType) {
```

Comment:
Could we maybe have `ClearDiagnostics` take in an optional parameter so that we don't have both the `ClearDiagnostics` and `ClearDiagnosticsByIssueType`? We have a lot of caching functions at the moment that are making it a bit hard to understand what's going on

Reply:
This might be a personal preference so feel free to disagree

Reply:
This looks pretty redundant, I agree.